### ISSUE TITLE
fn:elements-to-maps: Explicit Layouts

### ISSUE THREAD

**Issue by ChristianGruen** [qt4cg/qtspecs#1647]
If a user chooses a custom layout, it should always be applied, or (if inappropriate, by all means) an error message should be raised. 

The rationale: I think that the current fallback behavior is flawed. Explicit settings should never be overridden by implicit choices.

**Comment by michaelhkay** [comment]
I believe the current design does exactly that.

**Comment by ChristianGruen** [comment]
> I believe the current design does exactly that.

Sorry for ignoring that. Is it part of a PR? The [current spec](https://qt4cg.org/specifications/xpath-functions-40/Overview.html#id-element-layouts) still has rules like:

> **Layout:** `list`
> **Rules:** If the element has element children with names that are not all equal, or if it has non-whitespace text node children, then it is output as if mixed layout were chosen (see 17.4.1.9 Layout: Mixed). This is fallback behavior for use when this layout is chosen inappropriately.

**Comment by michaelhkay** [comment]
Well, it is using the rules for the requested layout: it's just that the rules for the requested layout say how list layout handles something that isn't a list.

If the user requests empty layout for an element that isn't empty, there are three possible things we can do: raise an error, produce output that drops the children, or use a different layout. I agree that the current rules aren't particularly consistent in deciding between these options. Nor is it obvious which of the three is best. I think it's best to avoid raising errors for this function; that means that every layout has to specify what happens to data that it wasn't designed for.

**Comment by ChristianGruen** [comment]
> I think it's best to avoid raising errors for this function

I agree for the automatic conversion, but I struggle to agree for explicitly chosen options. If someone expects a function to do something, I think we should either be obedient or speak out loud.

**Comment by michaelhkay** [comment]
The classic issue is of course the one described here and in many other posts:

https://stackoverflow.com/questions/79274373/

and the current design solves this by allowing you to select `{layouts: {'List': 'list'}}`.

**Comment by michaelhkay** [comment]
PR #1906 applies a consistent strategy to fallback when an inappropriate layout is chosen:

(a) if the layout does not allow attributes, then attributes are discarded
(b) in all other cases if an unappropriate layout is chosen, then the converter falls back to serialized XML layout, which will generally alert the user to the fact that the chosen layout could be improved; it also has the effect of not losing information.

**Comment by ChristianGruen** [comment]
Closed; see #1949 for updated feedback.
